app/components/progress_bar/index.tsx

```diff
@@ -24,50 +25,40 @@ const styles = StyleSheet.create({
 });
 
 const ProgressBar = ({color, progress, style}: ProgressBarProps) => {
-    const timer = useRef(new Animated.Value(progress)).current;
     const [width, setWidth] = useState(0);
 
-    useEffect(() => {
-        const animation = Animated.timing(timer, {
-            duration: 200,
-            useNativeDriver: true,
-            isInteraction: false,
-            toValue: progress,
-        });
+    const progressValue = useSharedValue(progress);
 
-        animation.start();
+    const progressAnimatedStyle = useAnimatedStyle(() => {
+        return {
+            transform: [
+                {translateX: withTiming(((progressValue.value * 0.5) - 0.5) * width, {duration: 200})},
+                {scaleX: withTiming(progressValue.value ? progressValue.value : 0.0001, {duration: 200})},
```

Comment:
this is fine, but we could've use interpolation here as well for simplicity. 0/5 if you want to change it, we can totally keep it.